ios/testmanagerd/xcuitestrunner.go

```diff
@@ -249,6 +250,40 @@ type TestConfig struct {
 	Listener *TestListener
 }
 
+func StartXCTestWithConfig(ctx context.Context, xctestrunFilePath string, testsToSkip []string, testsToRun []string, testArgs []string, testEnv map[string]interface{}, device ios.DeviceEntry, listener *TestListener) ([]TestSuite, error) {
+	// Parse the .xctestrun file to get the necessary details for TestConfig
+	codec := NewXCTestRunCodec()
+	result, err := codec.ParseFile(xctestrunFilePath)
+	if err != nil {
+		log.Errorf("Error parsing xctestrun file: %v", err)
+		return nil, err
+	}
+
+	// Verify that the FormatVersion is 1
+	if result.XCTestRunMetadata.FormatVersion != 1 {
+		log.Errorf("Invalid FormatVersion in .xctestrun file: got %d, expected 1", result.XCTestRunMetadata.FormatVersion)
+		return nil, fmt.Errorf("invalid FormatVersion in .xctestrun file: %d (expected 1)", result.XCTestRunMetadata.FormatVersion)
+	}
+
+	// Extract only the file name
+	var testBundlePath = filepath.Base(result.RunnerTests.TestBundlePath)
+
+	// Build the TestConfig object from parsed data
+	testConfig := TestConfig{
+		TestRunnerBundleId: result.RunnerTests.TestHostBundleIdentifier,
+		XctestConfigName:   testBundlePath,
+		Args:               testArgs,
+		Env:                testEnv,
+		TestsToRun:         testsToRun,
+		TestsToSkip:        testsToSkip,
+		XcTest:             true,
+		Device:             device,
+		Listener:           listener,
+	}
```

Comment:
`SkipTestIdentifiers` and `OnlyTestIdentifiers` are indeed in the [xctestrun](https://keith.github.io/xcode-man-pages/xcodebuild.xctestrun.5.html) but I believe we can't count on those yet since the  `FormatVersion` is 2